### PR TITLE
Upstream v5.46.0

### PR DESCRIPTION
**Pull request by t0yv0**
Upgrade upstream to v5.46.0

Fixes https://github.com/pulumi/pulumi-aws/issues/3847

### CODE REVIEW

File: patches/0027-Do-not-compute-tags_all-at-TF-level.patch [hunk 1]
```diff
@@ -4,23 +4,6 @@ Date: Mon, 6 Nov 2023 11:17:16 -0500
...
-diff --git a/internal/framework/base.go b/internal/framework/base.go
-index d1c4c7d1ab..3e51e53f87 100644
---- a/internal/framework/base.go
-+++ b/internal/framework/base.go
-@@ -70,6 +70,12 @@ func (r *ResourceWithConfigure) Configure(_ context.Context, request resource.Co
- 
- // SetTagsAll calculates the new value for the `tags_all` attribute.
- func (r *ResourceWithConfigure) SetTagsAll(ctx context.Context, request resource.ModifyPlanRequest, response *resource.ModifyPlanResponse) {
-+
-+	// Skip SetTagsAll in Pulumi because it is handled at Pulumi provider level.
```
Comment (t0yv0) [inline patches/0027-Do-not-compute-tags_all-at-TF-level.patch:16]: This needs a deeper look, we really relied on disabling SetTagsAll before.

Comment (t0yv0) [reply patches/0027-Do-not-compute-tags_all-at-TF-level.patch:16]: This is from

```
commit f7ab5f06d7387cc3cc1e292f47b562ac7a7bd822
Author: Kit Ewbank <Kit_Ewbank@hotmail.com>
Date:   Sat Oct 21 15:37:11 2023 -0400

    Move 'ResourceWithConfigure' to its own source file.
```

we need to match that.

Comment (t0yv0) [reply patches/0027-Do-not-compute-tags_all-at-TF-level.patch:16]: The file moved without changing. Matching it in the patch.

**Review by corymhall — APPROVED** [review]

**Review by flostadler — APPROVED** [review]

**Pull request merged**